File: examples/basic/test/unit/java/SayHelloStepTest.java
import com.lesfurets.jenkins.unit.BasePipelineTest;
import groovy.lang.Script;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class SayHelloStepTest extends BasePipelineTest {

    @BeforeEach
    void setup() throws Exception {
        setScriptRoots(new String[]{"."});
        setUp();
    }

    @Test
    void defaultGreeting() throws Exception {
        Script script = loadScript("vars/sayHello.groovy");
        script.invokeMethod("call", new Object[]{});
        assertTrue(getHelper().getCallStack().stream().anyMatch(c -> c.toString().contains("Hello, world!")));
    }

    @Test
    void namedGreeting() throws Exception {
        Script script = loadScript("vars/sayHello.groovy");
        script.invokeMethod("call", new Object[]{"Jenkins"});
        assertTrue(getHelper().getCallStack().stream().anyMatch(c -> c.toString().contains("Hello, Jenkins!")));
    }
}

File: examples/kotest/test/unit/kotlin/com/example/SayHelloStepTest.kt
package com.example

import com.lesfurets.jenkins.unit.BasePipelineTest
import io.kotest.core.spec.style.FunSpec
import io.kotest.inspectors.forAtLeastOne
import io.kotest.matchers.string.shouldContain
import org.codehaus.groovy.runtime.InvokerHelper

class SayHelloStepTest : FunSpec({
    lateinit var base: BasePipelineTest

    beforeEach {
        base =
            object : BasePipelineTest() {}.also { t ->
                t.setUp()
            }
    }

    test("default greeting") {
        InvokerHelper.invokeMethod(base.loadScript("vars/sayHello.groovy"), "call", null)
        base.helper.callStack.forAtLeastOne { it.toString() shouldContain "Hello, world!" }
    }

    test("named greeting") {
        InvokerHelper.invokeMethod(base.loadScript("vars/sayHello.groovy"), "call", "Jenkins")
        base.helper.callStack.forAtLeastOne { it.toString() shouldContain "Hello, Jenkins!" }
    }
})
